state/modelgeneration.go

```diff
@@ -171,24 +218,72 @@ func assignGenerationUnitTxnOps(id, appName, unitName string) []txn.Op {
 	}
 }
 
-// CanAutoComplete returns true if every application that has had configuration
+// MakeCurrent marks the generation as completed, if it is active and
+// meets autocomplete criteria, so it becomes the "current" generation.
+func (g *Generation) MakeCurrent() error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := g.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		if g.doc.Completed > 0 {
+			return nil, jujutxn.ErrNoOperations
+		}
+		if !g.Active() {
+			return nil, errors.New("generation is not currently active")
+		}
+		ok, err := g.CanMakeCurrent()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !ok {
+			return nil, errors.New("generation can not auto complete")
```

Comment:
Maybe just "generation can not be completed" or "generation can not be applied"?

Reply:
i went with the former